taca/utils/bioinfo_tab.py

```diff
@@ -0,0 +1,163 @@
+
+import socket
+import os
+import couchdb
+import glob
+import re
+import logging
+
+from csv import DictReader
+from taca.utils.config import CONFIG
+from flowcell_parser.classes import XTenSampleSheetParser
+
+logger = logging.getLogger(__name__)
+
+def setupServer(conf):
+    db_conf = conf['statusdb']
+    url="http://{0}:{1}@{2}:{3}".format(db_conf['username'], db_conf['password'], db_conf['url'], db_conf['port'])
+    return couchdb.Server(url)
+
+def merge(d1, d2):
+    """ Will merge dictionary d2 into dictionary d1.
+    On the case of finding the same key, the one in d1 will be used.
+    :param d1: Dictionary object
+    :param s2: Dictionary object
+    """
+    for key in d2:
+        if key in d1:
+            if isinstance(d1[key], dict) and isinstance(d2[key], dict):
+                merge(d1[key], d2[key])
+            elif d1[key] == d2[key]:
+                pass # same leaf value
```

Comment:
No, there used to be an else with logs, but I removed that because it was
pointless, so the elif becomes pointless too.

On Wed, Jul 8, 2015 at 5:06 PM, senthil10 <notifications@github.com> wrote:

> In taca/utils/bioinfo_tab.py
> <https://github.com/SciLifeLab/TACA/pull/134#discussion_r34158133>:
>
> > +    db_conf = conf['statusdb']
> > +    url="http://{0}:{1}@{2}:{3}".format(db_conf['username'], db_conf['password'], db_conf['url'], db_conf['port'])
> > +    return couchdb.Server(url)
> > +
> > +def merge(d1, d2):
> > +    """ Will merge dictionary d2 into dictionary d1.
> > +    On the case of finding the same key, the one in d1 will be used.
> > +    :param d1: Dictionary object
> > +    :param s2: Dictionary object
> > +    """
> > +    for key in d2:
> > +        if key in d1:
> > +            if isinstance(d1[key], dict) and isinstance(d2[key], dict):
> > +                merge(d1[key], d2[key])
> > +            elif d1[key] == d2[key]:
> > +                pass # same leaf value
>
> Any particular reason for this elif ? As all it does is passing
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/SciLifeLab/TACA/pull/134/files#r34158133>.
>